std/exception.d

```diff
@@ -1613,3 +1614,462 @@ version(unittest) package
+                }
+            }
+        }
+        else
+        {
+            @property bool empty()
+            {
+                return this.range.empty;
+            }
+        }
+
+        static if (isInputRange!IRange)
+        {
+            static if (functionsToHandle & RangePrimitive.front)
+            {
+                @property auto front()
+                {
+                    try
+                    {
+                        return this.range.front;
+                    }
+                    catch(E exception)
+                    {
+                        return handler(exception, this.range);
+                    }
+                }
+            }
+            else
+            {
+                @property auto front()
+                {
+                    return this.range.front;
+                }
+            }
+
+            static if (functionsToHandle & RangePrimitive.popFront)
+            {
+                void popFront()
+                {
+                    try
+                    {
+                        return this.range.popFront();
+                    }
+                    catch(E exception)
+                    {
+                        handler(exception, this.range);
+                    }
+                }
+            }
+            else
+            {
+                void popFront()
+                {
+                    this.range.popFront();
+                }
+            }
+        }
+
+        static if (isBidirectionalRange!IRange)
+        {
+            static if (functionsToHandle & RangePrimitive.back)
+            {
+                @property auto back()
+                {
+                    try
+                    {
+                        return this.range.back;
+                    }
+                    catch(E exception)
+                    {
+                        return handler(exception, this.range);
+                    }
+                }
+            }
+            else
+            {
+                @property auto back()
+                {
+                    return this.range.back;
+                }
+            }
+
+            static if (functionsToHandle & RangePrimitive.popBack)
+            {
+                void popBack()
+                {
+                    try
+                    {
+                        return this.range.popBack();
+                    }
+                    catch(E exception)
+                    {
+                        handler(exception, this.range);
+                    }
+                }
+            }
+            else
+            {
+                void popBack()
+                {
+                    this.range.popBack();
+                }
+            }
+        }
+
+        static if (isRandomAccessRange!IRange)
+        {
+            static if (is(typeof(range[ulong.max])))
+                private alias opIndex_t = ulong;
+            else
+                private alias opIndex_t = uint;
```

Comment:
OTOH, if the intent is to provide transparent forwarding of `opIndex`, perhaps we should be using ParameterTypeTuple to extract the exact argument type to `opIndex`?